UnityWeld_Editor/AnimatorParameterBindingEditor.cs

```diff
@@ -0,0 +1,337 @@
+            );
+
+            EditorStyles.label.fontStyle = viewAdapterPrefabModified ? FontStyle.Bold : defaultLabelStyle;
+
+            ShowAdapterMenu(
+                new GUIContent("View adapter", "Adapter that converts values sent from the view-model to the view."),
+                viewAdapterTypeNames,
+                targetScript.ViewAdapterTypeName,
+                newValue =>
+                {
+                    // Get rid of old adapter options if we changed the type of the adapter.
+                    if (newValue != targetScript.ViewAdapterTypeName)
+                    {
+                        Undo.RecordObject(targetScript, "Set view adapter options");
+                        targetScript.ViewAdapterOptions = null;
+                    }
+
+                    UpdateProperty(
+                        updatedValue => targetScript.ViewAdapterTypeName = updatedValue,
+                        targetScript.ViewAdapterTypeName,
+                        newValue,
+                        "Set view adapter"
+                    );
+                }
+            );
+
+            Type adapterType;
+            viewAdapterOptionsFade.target = ShouldShowAdapterOptions(targetScript.ViewAdapterTypeName, out adapterType);
+
+            EditorStyles.label.fontStyle = viewAdapterOptionsPrefabModified ? FontStyle.Bold : defaultLabelStyle;
+
+            ShowAdapterOptionsMenu(
+                "View adapter options",
+                adapterType,
+                options => targetScript.ViewAdapterOptions = options,
+                targetScript.ViewAdapterOptions,
+                viewAdapterOptionsFade.faded
+            );
+
+            EditorGUILayout.Space();
+
+            EditorStyles.label.fontStyle = viewModelPropertyPrefabModified ? FontStyle.Bold : defaultLabelStyle;
+
+            var adaptedViewPropertyType = AdaptTypeBackward(viewPropertyType, targetScript.ViewAdapterTypeName);
+            ShowViewModelPropertyMenu(
+                new GUIContent("View-model property", "Property on the view-model to bind to."),
+                TypeResolver.FindBindableProperties(targetScript),
+                updatedValue => targetScript.ViewModelPropertyName = updatedValue,
+                targetScript.ViewModelPropertyName,
+                property => property.PropertyType == adaptedViewPropertyType
+            );
+
+            GUI.enabled = guiPreviouslyEnabled;
+
+            EditorStyles.label.fontStyle = defaultLabelStyle;
+
+            EditorGUILayout.Space();
+
+        }
+
+        private void ShowAnimatorParametersMenu(
+            GUIContent label,
+            Action<AnimatorParameterTypeAndName> propertyValueSetter,
+            AnimatorParameterTypeAndName curPropertyValue,
+            out Type selectedPropertyType
+        )
+        {
+            var animator = targetScript.GetComponent<Animator>();
+
+            AnimatorControllerParameter[] properties = new AnimatorControllerParameter[] { };
```

Comment:
This doesn't need to be initialised to `new AnimatorControllerParameter[] { }` because it is assigned to again before it is ever used.